tools/automated_releaser/README.md

```diff
@@ -0,0 +1,55 @@
+# Automated GitHub Release Maker
+
+## Requirements
+
+- git
+- gh (github CLI)
+- Python
+- toml python library
+- Jinja2 python library
+
+To run the release automation:
+
+
+
+## Steps to Run
+1. Clone the library bundle and download all submodules. `clone_bundle.sh` can be used for this
+2. Copy the following files / folders into the root of the bundle:
+   - `release_maker_scripts/`
+   - `copy_release_making_files.py`
+   - `list_submodules.py`
+   - `remove_submodules.py`
```

Comment:
It may be simpler to package this up so it can be pip installed.

Reply:
Would you mind making it pip installable? That way you don't need to copy the script files around.

Reply:
I'm not sure being pip installable gets us away from needing to copy those files. 

The way it works now the user must copy those files into the bundle folder, and then to "launch"  the releaser part of the script is running `copy_release_making_files.py` which is then copying `release_maker_scripts/` into each repo within the bundle as it executes on each one. 

All of those scripts are currently written to assume they will be executed inside of each of those individual repos within the bundle.

In order to not require the user copying the files in it would have change everything that assumes it's running inside of that repo, or else we would need some other mechanism to write those files into each repo that is different from the `copy_release_making_files.py`

My experience with pip is fairly limited. Do you know if it's possible that code within a pip published library can write copies of the code within it into another file? So if `release_maker_scripts/` and it's contents are inside the pip published package, can it then write copies of all of those inside the specific repo it's working on for the current iteration?

Reply:
The other bit about publishing to pypi that I'm not sure about is whether it's possible to publish only a subsection of a repo to pip. Since this exists inside the Adabot repo, with the configuration that I'm familiar with it would deploy the full Adabot repo to pip, I've never done one that is only a specific subdirectory.

Reply:
@tannewt what if the copying didn't need to be done manually? one of the scripts (which I failed to add originally but have added with latest commit) is `clone_bundle.sh` which is what I use to clone and pull all submodules for the bundle. We could add some `cp` commands into that to copy the files into the bundle where the rest of the logic is currently expecting to find them.

I'm definitely up for tinkering with it to see if I can come up with a different way to do it that doesn't require copying the files in, but I don't have a clear idea of how to do it, so I can't promise a quick or even all the way successful solution. If I'm unable to get it working perhaps doing the copying inside the shell script is an acceptable alternative.

Reply:
I've never made one before but maybe it could be made to run with the `python -m` flag like `python -m http.server` That should still end up have the context and other files that it's expecting to find I think. 

The template that is currently a .md file could be changed to a string within a .py file so that it doesn't need to be opened with a file path. 

I think if that `-m` mechanism works how I am imagining it may offer a way to execute this without needing to copy the files in.